spec/controllers/verify/usps_controller_spec.rb

```diff
@@ -25,6 +26,37 @@
 
       expect(response).to be_ok
     end
+
+    it 'redirects if the user has sent too much mail' do
+      usps_mail_service = Idv::UspsMail.new(user)
+      allow(usps_mail_service).to receive(:mail_spammed?).and_return(true)
+      allow(subject.idv_session).to receive(:address_mechanism_chosen?).
+        and_return(true)
+      get :index
+
+      expect(response).to redirect_to verify_review_path
```

Comment:
There is a third option (the preferred one) that doesn't require writing to the DB or memoization: `instance_double`. We've been using this liberally in our tests. It looks like this:
```ruby
usps_mail_service = instance_double(Idv::UspsMail)
allow(Idv::UspsMail).to receive(:new).with(user).and_return(usps_mail_service)
allow(usps_mail_service).to receive(:mail_spammed?).and_return(true)
```